automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/FireCloudClient.scala

```diff
@@ -0,0 +1,113 @@
+package org.broadinstitute.dsde.firecloud.api
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model.headers.{Authorization, OAuth2BearerToken}
+import akka.http.scaladsl.model.{Multipart, _}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, _}
+import akka.util.ByteString
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.broadinstitute.dsde.firecloud.config.AuthToken
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutor, Future}
+
+class FireCloudClient {
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+
+  val mapper = new ObjectMapper()
+  mapper.registerModule(DefaultScalaModule)
+
+  // TODO: Is this okay here or should it be in a separate object and explicitly imported?
+  implicit class JsonStringUtil(s: String) {
+    def fromJsonMapAs[A](key: String): Option[A] = parseJsonAsMap.get(key)
+
+    def parseJsonAsMap[A]: Map[String, A] = mapper.readValue(s, classOf[Map[String, A]])
+  }
+
+  private def makeAuthHeader(token: AuthToken): Authorization = {
+    headers.Authorization(OAuth2BearerToken(token.value))
+  }
+
+  private def sendRequest(httpRequest: HttpRequest): HttpResponse = {
+    Await.result(Http().singleRequest(httpRequest), 5.minutes)
+  }
+
+  private def handleResponse(response: HttpResponse): String = {
+    response.status.isSuccess() match {
+      case true =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        Await.result(entityFuture, 100.millis).decodeString("UTF-8")
+      case _ =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        throw new APIException(Await.result(entityFuture, 100.millis).decodeString("UTF-8"))
+    }
+  }
+
+  def parseResponse(response: HttpResponse): String = {
```

Comment:
We lost some git history here, but I think code in `handleResponse` used to be inline in `requestWithJsonContent` or something like that.

@rtitle, the only issue with your proposed `sendRequestAndWaitForResponse` is that callers sometimes want to inspect the response rather than deal with one of a `String` response or an `APIException`. I like how it acts on futures, though, so I'm trying to work it into the API.